api/v1alpha1/raycluster_types.go

```diff
@@ -1,31 +1,83 @@
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
-// RayClusterSpec defines the desired state of RayCluster
+// RayClusterSpec defines the desired state of a RayCluster resource.
 type RayClusterSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Image used to launch head and worker nodes.
+	// +kubebuilder:default={repository: "ray-project/ray-ml", tag: "1.2.0-cpu"}
+	Image OCIImageDefinition `json:"image"`
+
+	// WorkerReplicaCount configures the total number of workers in the cluster.
+	// +kubebuilder:default=2
+	WorkerReplicaCount int `json:"workerReplicaCount"`
+
+	// HeadPort is the port of the head ray process.
+	// +kubebuilder:default=6379
+	HeadPort int `json:"port"`
+
+	// RedisShardPorts is a list of ports for non-primary Redis shards.
+	// +kubebuilder:default={6380,6381}
+	RedisShardPorts []int `json:"redisShardPorts"`
+
+	// ObjectManagerPort is the raylet port for the object manager.
+	// +kubebuilder:default=2384
+	ObjectManagerPort int `json:"objectManagerPort"`
+
+	// NodeManagerPort is the raylet port for the node manager.
+	// +kubebuilder:default=2385
+	NodeManagerPort int `json:"nodeManagerPort"`
+
+	// ObjectStoreMemoryBytes is amount of memory with which to start the object store.
+	ObjectStoreMemoryBytes int `json:"objectStoreMemoryBytes"`
+
+	// EnableDashboard starts the dashboard web UI.
+	// +kubebuilder:default=true
+	EnableDashboard bool `json:"enableDashboard"`
+
+	// DashboardPort is the port used by the dashboard server.
+	// +kubebuilder:default=8265
+	DashboardPort int `json:"dashboardPort"`
+
+	// EnableNetworkPolicy will create a network policy that restricts ingress to the cluster.
+	// +kubebuilder:default=true
+	EnableNetworkPolicy bool `json:"enableNetworkPolicy"`
+
+	// EnablePodSecurityPolicy will create a pod security policy that enforces secure execution of the ray process.
+	// +kubebuilder:default=true
+	EnablePodSecurityPolicy bool `json:"enablePodSecurityPolicy"`
+
+	// Labels additionally applied to cluster nodes.
+	Labels map[string]string `json:"labels"`
+
+	// Annotations applied to cluster nodes.
+	Annotations map[string]string `json:"annotations"`
+
+	// Resources requested and limits applied to cluster nodes.
+	// +kubebuilder:default={requests: {cpu: "100m", memory: "512Mi"}}
+	Resources corev1.ResourceRequirements `json:"resources"`
+
+	// Tolerations applied to cluster nodes.
+	Tolerations []corev1.Toleration `json:"tolerations"`
 
-	// Foo is an example field of RayCluster. Edit RayCluster_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// InitContainers added to cluster node pods.
+	InitContainers []corev1.Container `json:"initContainers"`
 }
```

Comment:
I'm already planning on adding an `emptyDir` volume mounted to `"/dev/shm"` without it having to be specified since that's what is recommended by the Ray peeps.